### ISSUE TITLE
rule.id should be optional if it matches the dictionary key name

### ISSUE THREAD

**Issue by lgolding** [oasis-tcs/sarif-spec#193]
@michaelcfanning Chris Meyer noticed this. There are two other cases (the `file` dictionary and the `logicalLocation` dictionary) where you can omit a property if it matches the dictionary key.

**Comment by michaelcfanning** [comment]
We should fix this for sure if these properties are documented currently as required. That's a bug in my opinion (in the sense that it is entirely unexpected if these properties are marked as required. they only exist to resolve key deduplication issues).

**Comment by lgolding** [comment]
Yes, the spec says that it's required (SHALL contain), and the schema agrees. Changed label from "design-improvement" to "bug". Non-breaking.
